service/svcerrors/user.go

```diff
@@ -0,0 +1,48 @@
+package svcerrors
+
+import "errors"
+
+var (
+	userNotFoundErrCode    = "user_not_found"
+	userUnknownErrCode     = "user_unknown_err"
+	userInvalidRoleErrCode = "invalid_user_role"
+)
+
+var (
+	ErrInvalidUserRole = SvcError{
+		code:    userInvalidRoleErrCode,
+		message: "invalid user role",
+		err:     errors.New("invalid user role"),
+	}
+)
+
+func NewUserErr(err error) error {
+	var repErr RepositoryError
+	if errors.As(err, &repErr) {
+		switch {
+		case repErr.IsNotFound():
+			return newUserNotFoundErr(err)
+		}
+	}
+
+	return newUserUnknownErr(err)
```

Comment:
If an error is not translated into a specific service error, I still want to return` <resource>UnknownError` to identify the service where the error occurred.